gitpod-io/gitpod
# [jetbrains] repo plugins support

## Description


This PR allows a user to specify JB plugins in .gitpod.yml, for instance: https://github.com/akosyakov/spring-petclinic/blob/8b4a927182ba75656c5d1fc3931482ea548206bd/.gitpod.yml#L33-L35

Unfortunately it is turned out that stable version of JB backend cannot support it. So this PR additionally introduces conditional building of JB backend plugin against stable and latest versions. For stable version support of repo plugins is disabled, and only enabled for EAP.

## Related Issue(s)


fix https://github.com/gitpod-io/gitpod/issues/6508

## How to test


#### Stable

- Make sure to **disable** latest version in settings: https://ak-jb-repo-plugins.staging.gitpod-dev.com/preferences
- Start a workspace on a repo with JB plugins with IntelliJ: https://ak-jb-repo-plugins.staging.gitpod-dev.com#referrer:jetbrains-gateway/https://github.com/akosyakov/spring-petclinic
- The client should start, but plugins won't be installed, since 2021.3 does not support headless mode properly and cannot install them from cli.

#### Latest
- Make sure to **enable** latest version in settings: https://ak-jb-repo-plugins.staging.gitpod-dev.com/preferences
- Start a workspace on a repo with JB plugins with IntelliJ: https://ak-jb-repo-plugins.staging.gitpod-dev.com#referrer:jetbrains-gateway/https://github.com/akosyakov/spring-petclinic
- The client should start and plugin should be installed.

## Release Notes

```release-note
Allow to configure JetBrains plugins in .gitpod.yml
```

## Documentation
